src/backend/glutin.rs

```diff
@@ -0,0 +1,359 @@
+    }
+
+    fn get_api(&self) -> Api {
+        self.context.get_api().into()
+    }
+
+    fn get_pixel_format(&self) -> PixelFormat {
+        self.context.get_pixel_format().into()
+    }
+}
+
+/// Window with an active Opengl Context created by `glutin`. Implements the
+/// `OpenglRenderer` graphics backend trait.
+pub struct GlutinWindowedRenderer
+{
+    window: Rc<Window>
+}
+
+impl GlutinWindowedRenderer
+{
+    fn new(window: Rc<Window>) -> GlutinWindowedRenderer {
+        GlutinWindowedRenderer {
+            window: window,
+        }
+    }
+}
+
+impl OpenglRenderer for GlutinWindowedRenderer
+{
+    #[inline]
+    fn swap_buffers(&self) -> Result<(), SwapBuffersError> {
+        match self.window.swap_buffers() {
+            Ok(()) => Ok(()),
+            Err(ContextError::IoError(e)) => panic!("Error while swapping buffers: {:?}", e),
+            Err(ContextError::ContextLost) => Err(SwapBuffersError::ContextLost),
+        }
+    }
+
+    #[inline]
+    unsafe fn get_proc_address(&self, symbol: &str) -> *const c_void {
+        self.window.get_proc_address(symbol) as *const _
+    }
+
+    #[inline]
+    fn get_framebuffer_dimensions(&self) -> (u32, u32) {
+        let (width, height) = self.window.get_inner_size().unwrap_or((800, 600));      // TODO: 800x600 ?
+        let scale = self.window.hidpi_factor();
+        ((width as f32 * scale) as u32, (height as f32 * scale) as u32)
+    }
+
+    #[inline]
+    fn is_current(&self) -> bool {
+        self.window.is_current()
+    }
+
+    #[inline]
+    unsafe fn make_current(&self) {
+        self.window.make_current().unwrap();
+    }
+
+    fn get_api(&self) -> Api {
+        self.window.get_api().into()
+    }
+
+    fn get_pixel_format(&self) -> PixelFormat {
+        self.window.get_pixel_format().into()
+    }
+}
+
+/// Errors that may happen when driving the event loop of `GlutinInputBackend`
+pub enum GlutinInputError
+{
+    /// The underlying `glutin` `Window` was closed. No further events can be processed.
+    ///
+    /// See `GlutinInputBackend::process_new_events`.
+    WindowClosed
+}
+
+/// Abstracted event loop of a `glutin` `Window` implementing the `InputBackend` trait
+///
+/// You need to call `process_new_events` periodically to receive any events.
+pub struct GlutinInputBackend
+{
+    window: Rc<Window>,
+    time_counter: u32,
+    seat: Seat,
+    handler: Option<Box<InputHandler + 'static>>,
+}
+
+impl InputBackend for GlutinInputBackend
+{
+    fn set_handler<H: InputHandler + 'static>(&mut self, mut handler: H) {
+        if self.handler.is_some() {
+            self.clear_handler();
+        }
+        handler.on_seat_created(&self.seat);
+        self.handler = Some(Box::new(handler));
+    }
+
+    fn get_handler(&mut self) -> Option<&mut InputHandler>
+    {
+        self.handler.as_mut().map(|handler| handler as &mut InputHandler)
+    }
+
+    fn clear_handler(&mut self) {
+        if let Some(ref mut handler) = self.handler {
```

Comment:
`if let Some(handler) = self.handler.take() { ... }` and removing the `self.handler = None;`